lsp/source/lib/util.mts

```diff
@@ -358,7 +358,7 @@ export function logTiming<R, A extends unknown[]>(name: string, fn: (...args: A)
     const start = performance.now(),
       result = fn(...args),
       end = performance.now();
-    console.log(`${name.padStart(32)}${(end - start).toFixed(2).padStart(8)}ms`)
+    // console.log(`${name.padStart(32)}${(end - start).toFixed(2).padStart(8)}ms`)
```

Comment:
Perhaps we can add a parameter or a global `logger` variable.

Reply:
Yes, I think so. A reasonable approach might be to add an LSP command-line parameter something like `--log server.log` that redirects all logs to there, and the default behavior remains going to `console` or `connection.console`.

Actually, if we can use `connection.console` always, presumably that can be intercepted? That seems like a cleaner solution.

Reply:
Indeed, having a universal behavior would indeed be better for interception. The problem here is that the required `connection` cannot be obtained.

This pull request will also address this problem.